extension-workspace-reference.md

```diff
@@ -0,0 +1,111 @@
+# workspace/reference extension to LSP
+
+The `workspace/reference` extension to the Language Server Protocol (LSP) enables a language server to export all of the references that code inside of the workspace root makes to code symbols that are external to the workspace root (e.g. dependencies).
```

Comment:
I would agree with you (in fact, that was the original name), but `workspace/symbol` isn't `workspace/symbols`.. so I think it should be named this way. Let me know if you can think of a compelling reason otherwise and I'll change it!